pkg/cmd/roachtest/roachtestutil/httpclient.go

```diff
@@ -134,6 +134,9 @@ func (r *RoachtestHTTPClient) addCookies(ctx context.Context, cookieUrl string)
 	if !r.cluster.IsSecure() {
 		return nil
 	}
+
+	r.mu.Lock()
+	defer r.mu.Unlock()
```

Comment:
Nice find! Because of this data race, there was a high bias to return `n1`; now, all other nodes are equally likely to be picked (in `postTestAssertions`).